### ISSUE TITLE
Auto-Login failing

### ISSUE THREAD

**Issue by Plaristote** [LeonardoRM/owncloud-roundcube#10]
Greetings.

I'm using owncloud 10.7.0, and fetched this plugin from git a couple of days ago.

I maanged to get the roundcube instance to open in an iframe, but that's pretty much all that's working.

I know that the extension successfully authenticate to Roundcube: I've checked that the credentials used by `BackLogin` were correct, and I've also noticed that if I have a session opened with Roundcube on another tab, it expires whenever I start a new session from Owncloud.

The following logs appear when I open the Roundcube app from Owncloud:

```
{"reqId":"F2BXjHcb888NcFaStJ6Y","level":3,"time":"June 06, 2021 20:04:35","remoteAddr":"10.0.86.238","user":"plaristote","app":"PHP","method":"POST","url":"\/index.php\/apps\/roundcube\/ajax\/adminSettings.php","message":"Undefined variable: path at \/var\/www\/owncloud\/apps-external\/roundcube\/lib\/Controller\/SettingsController.php#75"}
{"reqId":"F2BXjHcb888NcFaStJ6Y","level":3,"time":"June 06, 2021 20:04:35","remoteAddr":"10.0.86.238","user":"plaristote","app":"PHP","method":"POST","url":"\/index.php\/apps\/roundcube\/ajax\/adminSettings.php","message":"Invalid argument supplied for foreach() at \/var\/www\/owncloud\/apps-external\/roundcube\/lib\/Controller\/SettingsController.php#80"}
{"reqId":"F2BXjHcb888NcFaStJ6Y","level":3,"time":"June 06, 2021 20:04:35","remoteAddr":"10.0.86.238","user":"plaristote","app":"PHP","method":"POST","url":"\/index.php\/apps\/roundcube\/ajax\/adminSettings.php","message":"Invalid argument supplied for foreach() at \/var\/www\/owncloud\/apps-external\/roundcube\/lib\/Controller\/SettingsController.php#88"}
{"reqId":"F2BXjHcb888NcFaStJ6Y","level":3,"time":"June 06, 2021 20:04:35","remoteAddr":"10.0.86.238","user":"plaristote","app":"PHP","method":"POST","url":"\/index.php\/apps\/roundcube\/ajax\/adminSettings.php","message":"count(): Parameter must be an array or an object that implements Countable at \/var\/www\/owncloud\/apps-external\/roundcube\/lib\/Controller\/SettingsController.php#101"}
{"reqId":"F2BXjHcb888NcFaStJ6Y","level":3,"time":"June 06, 2021 20:04:35","remoteAddr":"10.0.86.238","user":"plaristote","app":"PHP","method":"POST","url":"\/index.php\/apps\/roundcube\/ajax\/adminSettings.php","message":"count(): Parameter must be an array or an object that implements Countable at \/var\/www\/owncloud\/apps-external\/roundcube\/lib\/Controller\/SettingsController.php#101"}
{"reqId":"F2BXjHcb888NcFaStJ6Y","level":3,"time":"June 06, 2021 20:04:35","remoteAddr":"10.0.86.238","user":"plaristote","app":"PHP","method":"POST","url":"\/index.php\/apps\/roundcube\/ajax\/adminSettings.php","message":"array_combine() expects parameter 1 to be array, string given at \/var\/www\/owncloud\/apps-external\/roundcube\/lib\/Controller\/SettingsController.php#117"}
{"reqId":"F2BXjHcb888NcFaStJ6Y","level":3,"time":"June 06, 2021 20:04:35","remoteAddr":"10.0.86.238","user":"plaristote","app":"PHP","method":"POST","url":"\/index.php\/apps\/roundcube\/ajax\/adminSettings.php","message":"array_filter() expects parameter 1 to be array, null given at \/var\/www\/owncloud\/apps-external\/roundcube\/lib\/Controller\/SettingsController.php#121"}
{"reqId":"CHAe6kjLLMkmG4vtyl5Q","level":3,"time":"June 06, 2021 20:04:39","remoteAddr":"10.0.86.238","user":"plaristote","app":"PHP","method":"GET","url":"\/index.php\/apps\/roundcube\/","message":"Undefined index: set-cookie at \/var\/www\/owncloud\/apps-external\/roundcube\/lib\/BackLogin.php#76"}
{"reqId":"CHAe6kjLLMkmG4vtyl5Q","level":3,"time":"June 06, 2021 20:04:39","remoteAddr":"10.0.86.238","user":"plaristote","app":"PHP","method":"GET","url":"\/index.php\/apps\/roundcube\/","message":"Undefined index: _token at \/var\/www\/owncloud\/apps-external\/roundcube\/lib\/BackLogin.php#83"}
{"reqId":"CHAe6kjLLMkmG4vtyl5Q","level":3,"time":"June 06, 2021 20:04:39","remoteAddr":"10.0.86.238","user":"plaristote","app":"PHP","method":"GET","url":"\/index.php\/apps\/roundcube\/","message":"Trying to access array offset on value of type null at \/var\/www\/owncloud\/apps-external\/roundcube\/lib\/BackLogin.php#83"}
{"reqId":"CHAe6kjLLMkmG4vtyl5Q","level":3,"time":"June 06, 2021 20:04:39","remoteAddr":"10.0.86.238","user":"plaristote","app":"PHP","method":"GET","url":"\/index.php\/apps\/roundcube\/","message":"Undefined index: _timezone at \/var\/www\/owncloud\/apps-external\/roundcube\/lib\/BackLogin.php#86"}
{"reqId":"CHAe6kjLLMkmG4vtyl5Q","level":3,"time":"June 06, 2021 20:04:39","remoteAddr":"10.0.86.238","user":"plaristote","app":"PHP","method":"GET","url":"\/index.php\/apps\/roundcube\/","message":"Trying to access array offset on value of type null at \/var\/www\/owncloud\/apps-external\/roundcube\/lib\/BackLogin.php#86"}
{"reqId":"CHAe6kjLLMkmG4vtyl5Q","level":3,"time":"June 06, 2021 20:04:39","remoteAddr":"10.0.86.238","user":"plaristote","app":"PHP","method":"GET","url":"\/index.php\/apps\/roundcube\/","message":"Undefined index: _url at \/var\/www\/owncloud\/apps-external\/roundcube\/lib\/BackLogin.php#87"}
{"reqId":"CHAe6kjLLMkmG4vtyl5Q","level":3,"time":"June 06, 2021 20:04:39","remoteAddr":"10.0.86.238","user":"plaristote","app":"PHP","method":"GET","url":"\/index.php\/apps\/roundcube\/","message":"Trying to access array offset on value of type null at \/var\/www\/owncloud\/apps-external\/roundcube\/lib\/BackLogin.php#87"}
{"reqId":"CHAe6kjLLMkmG4vtyl5Q","level":3,"time":"June 06, 2021 20:04:39","remoteAddr":"10.0.86.238","user":"plaristote","app":"PHP","method":"GET","url":"\/index.php\/apps\/roundcube\/","message":"Undefined index: set-cookie at \/var\/www\/owncloud\/apps-external\/roundcube\/lib\/BackLogin.php#98"}
```

Seems like `BackLogin` is having some issues setting up the cookies.

On that regard, the fact that everything happens behind a proxy might be relevant ?

Anyhow, I'll see what I can do, and try to provide a patch if necessary, but some guidance would be much appreciated !

**Comment by Plaristote** [comment]
After further inspection, I found out the last issue was caused by roundcube returning 301 to the first login request. I was able to fix the issue by setting CURLOPT_FOLLOWLOCATION to true in `BackLogin::sendRequest`. 

Another issue is now preventing autologin again. I'm using Roundcube 1.4.6, and I'm starting to think that version might be unsupported ?

Trying to access the roundcube app now results in a 504 error. Seems like some requests are timing out. It might be related to other 301 responses from Roundcube ? That's what I'll be checking out next.

Here are the logs, debug level this time:

```
{"reqId":"tlyJYzOBbcxuH2skdeMg","level":0,"time":"June 06, 2021 21:00:25","remoteAddr":"10.0.86.238","user":"plaristote","app":"roundcube","method":"GET","url":"\/index.php\/apps\/roundcube\/","message":"OCA\\RoundCube\\BackLogin::sendRequest: URL: 'https:\/\/cloud.planed.es\/roundcube?_task=login'."}
{"reqId":"tlyJYzOBbcxuH2skdeMg","level":0,"time":"June 06, 2021 21:00:26","remoteAddr":"10.0.86.238","user":"plaristote","app":"roundcube","method":"GET","url":"\/index.php\/apps\/roundcube\/","message":"OCA\\RoundCube\\BackLogin::sendRequest: Got the following HTTP Status Code: (200) 0: "}
{"reqId":"tlyJYzOBbcxuH2skdeMg","level":0,"time":"June 06, 2021 21:00:26","remoteAddr":"10.0.86.238","user":"plaristote","app":"roundcube","method":"GET","url":"\/index.php\/apps\/roundcube\/","message":"OCA\\RoundCube\\BackLogin::login: Post Login form: 'plaristote' 'SecretPassword123'."}
{"reqId":"tlyJYzOBbcxuH2skdeMg","level":0,"time":"June 06, 2021 21:00:26","remoteAddr":"10.0.86.238","user":"plaristote","app":"roundcube","method":"GET","url":"\/index.php\/apps\/roundcube\/","message":"OCA\\RoundCube\\BackLogin::sendRequest: URL: 'https:\/\/cloud.planed.es\/roundcube?_task=login&_action=login'."}
{"reqId":"tlyJYzOBbcxuH2skdeMg","level":0,"time":"June 06, 2021 21:01:26","remoteAddr":"10.0.86.238","user":"plaristote","app":"roundcube","method":"GET","url":"\/index.php\/apps\/roundcube\/","message":"OCA\\RoundCube\\BackLogin::sendRequest: Got the following HTTP Status Code: (301) 28: Operation timed out after 59999 milliseconds with 0 bytes received"}
{"reqId":"tlyJYzOBbcxuH2skdeMg","level":2,"time":"June 06, 2021 21:01:26","remoteAddr":"10.0.86.238","user":"plaristote","app":"roundcube","method":"GET","url":"\/index.php\/apps\/roundcube\/","message":"OCA\\RoundCube\\BackLogin::sendRequest: Opening url 'https:\/\/cloud.planed.es\/roundcube?_task=login&_action=login' failed with 'Operation timed out after 59999 milliseconds with 0 bytes received'"}
{"reqId":"tlyJYzOBbcxuH2skdeMg","level":3,"time":"June 06, 2021 21:01:26","remoteAddr":"10.0.86.238","user":"plaristote","app":"roundcube","method":"GET","url":"\/index.php\/apps\/roundcube\/","message":"OCA\\RoundCube\\BackLogin::login: Could not get login response."}
```

**Comment by Plaristote** [comment]
Wow... after all the time spent on this, I almost fainted when, at long last, I had my hands on my holy grail. It works !

I've opened PR #11 and #12 to implement the changes that helped me getting this application up and running.

I also have a couple of issues with settings:
- I had to hardcode the targets for roundcube queries, as the value set in the settings page didn't seem to be used anywhere (or was it ? I'm still very new to owncloud, I can't really tell).
- The settings to remove Roundcube's infobar does not seem to work either. I'm shooting in the dark here, but that might be because I'm using Roundcube 1.4.6 with the elastic theme.

But for now, it seems that autologin's good to go, so I'm closing this issue.
